azurerm/internal/services/network/virtual_hub_resource.go

```diff
@@ -142,10 +144,23 @@ func resourceArmVirtualHubCreateUpdate(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("Error waiting for creation of Virtual Hub %q (Resource Group %q): %+v", name, resourceGroup, err)
 	}
 
-	resp, err := client.Get(ctx, resourceGroup, name)
+	// Hub returns provisioned while the routing state is still "provisining". This might cause issues with following hubvnet connection operations.
+	// https://github.com/Azure/azure-rest-api-specs/issues/10391
+	// As a workaround, we will poll the routing state and ensure it is "Provisioned".
+	timeout, _ := ctx.Deadline()
```

Comment:
all timeouts should be caught, but we should still raise the error here

Reply:
The second returned value is a `bool` indicating whether the deadline is set on the context, as we have that guaranteed via CI/Makefile, we can simply ignore handling it here?
(did you treat `ctx.Deadline()` as `ctx.Err()` by mistake?)

Reply:
that should be trie - but isn't necessarily if we end up wrapping this in a migration function for whatever reason (where we'd likely use `context.TODO()`, as I believe we do for storage accounts - so there's no harm in checking/raising this, even if it'd be a niche use-case/error

Reply:
OK, I'll put the check at the very begining of this function to fail fast (if deadline is not set properly).
(does *niche* mean nicht 🇩🇪 here)